macros/blueprint-proc-macro/src/job.rs

```diff
@@ -742,15 +770,11 @@ impl Parse for JobArgs {
 
         let id = id.ok_or_else(|| input.error("Missing `id` argument in attribute"))?;
 
-        if params.is_empty() {
-            return Err(input.error("Missing `params` argument in attribute"));
-        }
-
-        let result = result.ok_or_else(|| input.error("Missing 'result' argument in attribute"))?;
+        let result = result.unwrap_or(ResultsKind::Infered);
 
         if let ResultsKind::Types(ref r) = result {
             if r.is_empty() {
-                return Err(input.error("Expected at least one parameter for the `result` attribute, or `_` to infer the type"));
+                return Err(input.error("Expected at least one parameter for the `result` attribute, or `_` to infer the type, or nothing to infer the type from the function return type"));
```

Comment:
```suggestion
                return Err(input.error("`result` attribute empty, expected at least one parameter, or `_` to infer the type"));
```
The error is getting long, once #381 is being worked on we can use a diagnostic crate and provide that as a hint.